### PR TITLE
[IMP] website: improve reset view feature

### PR DESCRIPTION
**Pull request by rdeodoo**
- Store previous arch to be able to reset it (soft reset)
- Add the possibility to reset from file if possible (hard reset)
- Adapt frontend reset page to these new fields
- `arch_fs` hack to check if view was modified got moved to new field
  `arch_updated` as we now need to keep track of the `arch_fs` to reset a
  broken view.

Closes #32009 (task-1943001)

This commit also:
- adds the possibility to reset a view arch in its form view, including a diff
  viewer (github like).
- adds string helper on those arch fields, as they are not documented, it is not
  always obvious to get the purpose of each fields.
- adapts the font-size and headings to fit Odoo style on the error 500 page.
  The error page does not use Odoo's assets, just bootstrap.css (the cursor
  might be broken on the error 500 page so we can't use Odoo assets).
- raise the caller on unexisting t-call during QWeb rendering.
- adapts reset view code to single view as the code still had remains of its
  initial release (Odoo 9.0). Handling multiple views has no sense anymore.

### CODE REVIEW

**Comment by rdeodoo** [comment]
2 backport commits to decide, they probably can/should go in 12.0
Commit to remove dev purpose can be removed before merge
Commit to remove cherry pick wait for cherrypick, got to ensure it is forward ported like that commit (it will conflict, xpath needed)

let me know if I can help your team to finish this

**Comment by rdeodoo** [comment]
Rebased, needed commit hit master.
left to do:
1. Decide about the 2 commits prefixed by BACKPORT if we merge those in 12.0 or master (I prefer 12.0)
2. Write some tests (we can do it after merge but I'd prefer to do it before)

**Comment by rdeodoo** [comment]
> * no backport

Ok
> * typo in commit msg ([6dcad0d](https://github.com/odoo/odoo/commit/6dcad0d2fb8040d8f705381939878210b5eec6f3) Qwew and second § seems wrong)

Done
> * Some extra comment to remove in code ([4978305](https://github.com/odoo/odoo/commit/497830539530e45b38c5f96201cc0c27661f3435))

Done
> * Commit in strange order, second commit use field from next commit

Not sure what field? Order seems good on the PR:
1. Improve (new field)
2.  Add wizard in backend
3. Qweb return correct view when unknown t-call
> * Tooltip for arch_fs is too verbose Used for reset or dev mode. Not need to explain all the code

I am ok to reduce from
```
File from where the view originates.
Useful to (hard) reset broken views or to read arch from file in dev-xml mode.
Modifying the view arch will empty this field
```
to
```
File from where the view originates.
Used for view reset or dev-xml mode.
Modifying the view arch will empty this field.
```
Let me know if you absolutely want me to remove the last part `Modifying the view arch will empty this field.` but I think it is not intuitive that that field got removed when writing on `arch`.
I worked on every `help` from all the `arch_xx` fields so you can fully understand why they need to exists.

> * Use at least css instead of style inline for 500 layout

Done, replaced
```xml
<html style="font-size: 14px;">
```
by `style` block on the 500 template:
```xml
<style>
html{
    font-size: 14px;
}
</style>
```
>   To check: apply it only for Qweb view; we don't need it to duplicate all views for form view and co

Yes, that was one of the point still up to discuss on the task. If you are ok too with that let's do it only for QWeb views.
> * About arch_initial and arch_fs, probably missing something, but why not keep arch_fs and just flag boolean 'arch_updated' instead to copy the arch intial in case you loose the arch_fs. In this case you will be able to read initial in all case, and restore add the missing feature:: restore from file.
> * I didn't see the last step: restore from XML that we speak about. (usefull when you want restore the last commited view)

`arch_fs` is only related to data/demo views, not for oe_structure and new pages. I'd like to further debate this.

**Comment by rdeodoo** [comment]
As decided, `arch_initial` was removed.
hard reset mode now reset to file view, meaning only file views will have that possibility, eg not created website.page.
COW view will be able to reset to file view as `arch_fs` is no more erased on `arch` update. That hack to prefer modified arch over file arch got moved to new field `arch_updated`.

Also, as `reset_templates` could only reset one view only (the old implementation < odoo 9.0 returned the all views_get tree which is not the case anymore), I adapted everything, routes, xml, js..

LibDiff table is now totally responsive, I added `<style>` tag for remaining finetunning, this means we got rid of the weird `wrapcolumn`.

Change arch order on diff view so it is like in github, new diff t othe right and old diff to the left

I left those changes in a separate commit to amend so it remains encapsulated for the review.

**Comment by rdeodoo** [comment]
I also added a test for the hard reset and adapter test methods (`fix_it`, `find_template`) to new DOM

File: odoo/addons/base/models/ir_ui_view.py [hunk 1]
```diff
@@ -205,10 +206,20 @@ class View(models.Model):
...
+    arch_base = fields.Text(compute='_compute_arch_base', inverse='_inverse_arch_base', string='Base View Architecture',
+                            help="This field is the same as `arch` field without translations")
+    arch_db = fields.Text(string='Arch Blob', translate=xml_translate, oldname='arch',
+                          help="This field stores the view arch.")
+    arch_fs = fields.Char(string='Arch Filename', help="""File from where the view originates.
+                                                          Useful to (hard) reset broken views or to read arch from file in dev-xml mode.""")
+    arch_updated = fields.Boolean(string='Modified Architecture')
+    arch_prev = fields.Text(string='Previous View Architecture', help="""This field will save the current `arch_db` before writing on it.
+                                                                         Useful to (soft) reset a broken view.""")
+    arch_initial = fields.Text(string='Initial View Architecture', help="""This field will save the initial `arch_db` when the view is created.
```
Comment (JKE-be) [inline odoo/addons/base/models/ir_ui_view.py]: can we remove it now?

Comment (rdeodoo) [reply odoo/addons/base/models/ir_ui_view.py]: Whoops, yes it was supposed to be removed, I removed every occurences and uses of this field but not the field, done.

**Comment by rdeodoo** [comment]
Title/Desc of the PR udpated

**Comment by JKE-be** [comment]
@robodoo r+

//cc @d-fence

**Comment by d-fence** [comment]
@robodoo rebase-merge

**Pull request closed without merging**